### PR TITLE
Replace ucte-x-node-code with pairing_key

### PR DESCRIPTION
**Pull request by colineplqt**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Update

**What is the current behavior?**

Dangling lines have a field ucte-x-node-code

**What is the new behavior (if this is a feature change)?**
This field is renamed pairing_key and ucte-x-node-code is deprecated

**Does this PR introduce a breaking change or deprecate an API?**
- [x] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

When creating a dangling line, the column or keyword argument `ucte-x-node-code` should be replaced by `pairing_key`.
When accessing the field from the dangling dataframe, the name of the column is also updated to `pairing_key`.

### CODE REVIEW

File: pypowsybl/network/impl/network.py [hunk 1]
```diff
@@ -3619,6 +3620,16 @@ def create_dangling_lines(self, df: DataFrame = None, **kwargs: ArrayLike) -> No
                 network.create_dangling_lines(id='BAT-1', voltage_level_id='VL1', bus_id='B1',
                                               p0=10, q0=3, r=0, x=5, g=0, b=1e-6)
         """
+        if df is not None:
```
Comment (colineplqt) [inline pypowsybl/network/impl/network.py:3625]: I need a review on that, I don't know if other ways to deprecate should be used.

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:3625]: maybe you can try with a test

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:3625]: I'll do it if you can't run test

Comment (colineplqt) [reply pypowsybl/network/impl/network.py:3625]: I tested in a jupyter notebook and it worked but I did not write any tests as I can't manage to make them run

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:3625]: ok I test

Comment (EtienneLt) [reply pypowsybl/network/impl/network.py:3625]: @geofjamg @annetill  are we doing a breaking change or we maintain compatibility (can be much code for not so much)

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**